### ISSUE TITLE
fit_gpytorch_mll fails on cuda device

### ISSUE THREAD

**Issue by Balandat** [pytorch/botorch#1566]
```
from botorch.fit import fit_gpytorch_mll
fit_gpytorch_mll(mll);
---------------------------------------------------------------------------
MDNotImplementedError                     Traceback (most recent call last)
[/usr/local/lib/python3.8/dist-packages/botorch/utils/dispatcher.py](https://localhost:8080/#) in __call__(self, *args, **kwargs)
     92         try:
---> 93             return func(*args, **kwargs)
     94         except MDNotImplementedError:

8 frames
MDNotImplementedError: 

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
[/usr/local/lib/python3.8/dist-packages/torch/_tensor.py](https://localhost:8080/#) in __array__(self, dtype)
    955             return self.numpy()
    956         else:
--> 957             return self.numpy().astype(dtype, copy=False)
    958 
    959     # Wrap Numpy array again in a suitable tensor when done, to support e.g.

TypeError: can't convert cuda:0 device type tensor to numpy. Use Tensor.cpu() to copy the tensor to host memory first.
```

which disappears after I move mll to CPU:
`fit_gpytorch_mll(mll.cpu());`

The model is in GPU:
`print(next(model.parameters()).is_cuda)
True`

_Originally posted by @jackliu333 in https://github.com/pytorch/botorch/issues/1323#issuecomment-1352487057_

**Comment by jackliu333** [comment]
Sure, please find below reproducible example:
```
import torch

device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
dtype = torch.double

# unknown objective function
from botorch.test_functions import Hartmann
neg_hartmann6 = Hartmann(negate=True)

# generate initial dataset
NOISE_SE = 0.1

def generate_initial_data(n=10):
    # generate training data
    train_x = torch.rand(n, 6, device=device, dtype=dtype)
    exact_obj = neg_hartmann6(train_x).unsqueeze(-1)  # add output dimension
    train_y = exact_obj + NOISE_SE * torch.randn_like(exact_obj)
    best_observed_value = train_y.max().item()
    return train_x, train_y, best_observed_value

train_x, train_y, best_observed_value = generate_initial_data(n=5)

# initialize GP model
from botorch.models import SingleTaskGP
from gpytorch.mlls import ExactMarginalLogLikelihood

def initialize_model(train_x, train_y):
    # create a single-task exact GP model instance
    # use a GP prior with Matern kernel and constant mean function by default
    model = SingleTaskGP(train_X=train_x, train_Y=train_y)
    mll = ExactMarginalLogLikelihood(model.likelihood, model)

    return mll, model

mll, model = initialize_model(train_x, train_y)

print(next(model.parameters()).is_cuda)

# fit GP hyperparameters - works
print("moving to cpu to optimize GP hyperparameters")
from botorch.fit import fit_gpytorch_model
# fit hyperparameters (kernel parameters and noise variance) of a GPyTorch model 
fit_gpytorch_model(mll.cpu());
mll = mll.to(train_x)
model = model.to(train_x)

# fit GP hyperparameters - error
print("hit error when running in gpu")
fit_gpytorch_model(mll);
```
Output:
```
True
moving to cpu to optimize GP hyperparameters
hit error when running in gpu
---------------------------------------------------------------------------
MDNotImplementedError                     Traceback (most recent call last)
[/usr/local/lib/python3.8/dist-packages/botorch/utils/dispatcher.py](https://localhost:8080/#) in __call__(self, *args, **kwargs)
     92         try:
---> 93             return func(*args, **kwargs)
     94         except MDNotImplementedError:

9 frames
MDNotImplementedError: 

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
[/usr/local/lib/python3.8/dist-packages/torch/_tensor.py](https://localhost:8080/#) in __array__(self, dtype)
    955             return self.numpy()
    956         else:
--> 957             return self.numpy().astype(dtype, copy=False)
    958 
    959     # Wrap Numpy array again in a suitable tensor when done, to support e.g.

TypeError: can't convert cuda:0 device type tensor to numpy. Use Tensor.cpu() to copy the tensor to host memory first.
```

**Comment by Balandat** [comment]
Thanks for the repro - this runs fine on our end though. What is the botorch version you're using (also the gpytorch and torch versions are of interest).

**Comment by saitcakmak** [comment]
@jackliu333 This should be fixed by #1567.
